src/tests/ftest/deployment/server_rank_failure.py

```diff
@@ -0,0 +1,375 @@
+        self.server_managers[0].restart(hosts=self.hostlist_servers)
+
+    def kill_engine(self, engine_kill_host):
+        """Kill engine on the given host.
+
+        Args:
+            engine_kill_host (str): Hostname to kill engine.
+        """
+        pattern = self.server_managers[0].manager.job.command_regex
+        result = stop_processes(hosts=[engine_kill_host], pattern=pattern)
+        if 0 in result and len(result) == 1:
+            self.log.info("No remote daos_engine process killed!")
+        else:
+            self.log.info("daos_engine in %s killed", engine_kill_host)
+
+    def verify_ior_worked(self, ior_results, job_num, errors):
+        """Verify that the IOR worked.
+
+        Args:
+            ior_results (dict): Dictionary that contains the IOR run results.
+            job_num (int): Job number used for the IOR run.
+            errors (list): Error list used in the test.
+        """
+        self.log.info(ior_results)
+        if not ior_results[job_num][0]:
+            ior_error = ior_results[job_num][1]
+            errors.append("Error found in second IOR run! {}".format(ior_error))
+
+    def verify_rank_failure(self, ior_namespace, container_namespace):
+        """Verify engine failure can be recovered by restarting daos_server.
+
+        1. Create a pool and a container. Create a container with or without redundancy
+        factor based on container_namespace.
+        2. Run IOR with given object class.
+        3. While IOR is running, kill daos_engine processes from one of the nodes to
+        simulate a node failure. (Two engines per node.)
+        4. Verify that IOR failed.
+        5. Restart daos_server service on the two nodes.
+        6. Verify the system status by calling dmg system query.
+        7. Verify that the container Health is HEALTHY.
+        8. Run IOR to the same container and verify that it works.
+
+        Args:
+            ior_namespace (str): Yaml namespace that defines the object class used for
+                IOR.
+            container_namespace (str): Yaml namespace that defines the container
+                redundancy factor.
+        """
+        # 1. Create a pool and a container.
+        self.add_pool(namespace="/run/pool_size_ratio_80/*")
+        self.add_container(pool=self.pool, namespace=container_namespace)
+
+        # 2. Run IOR.
+        ior_results = {}
+        job_num = 1
+        # If we don't use timeout, when the engines are killed, Mpirun gets stuck and
+        # waits forever. If we use timeout that's too long, the daos container command
+        # after the server restart will get stuck, so use 10 sec timeout, which is the
+        # same as deadlineForStonewalling in IOR.
+        ior_thread = threading.Thread(
+            target=self.run_ior_report_error,
+            args=[ior_results, job_num, "test_file_1", self.pool, self.container,
+                  ior_namespace, 10])
```

Comment:
Could we put this timeout in the yaml? Other test cases use, e.g.
```
ior:
    ior_timeout: 120
```
Or, we could just use `sw_deadline` here

Reply:
This is the Mpirun timeout value. If we don't set it, and just use `sw_deadline: 10`, and kill the engine, Mpirun gets stuck, so we want Mpirun to end as well in that case.

Reply:
Since we want the Mpirun timeout to match sw_deadline, I was thinking we could do something like this:
```
mpirun_timeout = self.params.get('sw_deadline', ior_namespace)
```

Reply:
Done. Thanks.